### ISSUE TITLE
marvin setup-openai fails with "cannot find context for fork"

### ISSUE THREAD

**Issue by ccrowner** [PrefectHQ/marvin#107]
Traceback (most recent call last):
  File "C:\Users\ccrow\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\ccrow\AppData\Local\Programs\Python\Python310\lib\runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "C:\Users\ccrow\AppData\Local\Programs\Python\Python310\Scripts\marvin.exe\__main__.py", line 4, in <module>
  File "C:\Users\ccrow\AppData\Local\Programs\Python\Python310\lib\site-packages\marvin\__init__.py", line 18, in <module>
    from marvin.utilities.logging import get_logger
  File "C:\Users\ccrow\AppData\Local\Programs\Python\Python310\lib\site-packages\marvin\utilities\__init__.py", line 1, in <module>
    from . import logging, async_utils, types, strings, collections, models
  File "C:\Users\ccrow\AppData\Local\Programs\Python\Python310\lib\site-packages\marvin\utilities\async_utils.py", line 10, in <module>
    process_pool = concurrent.futures.ProcessPoolExecutor(mp_context=mp.get_context("fork"))
  File "C:\Users\ccrow\AppData\Local\Programs\Python\Python310\lib\multiprocessing\context.py", line 243, in get_context
    return super().get_context(method)
  File "C:\Users\ccrow\AppData\Local\Programs\Python\Python310\lib\multiprocessing\context.py", line 193, in get_context
    raise ValueError('cannot find context for %r' % method) from None
ValueError: cannot find context for 'fork'

**Comment by ccrowner** [comment]
seems this may be because I'm running Windows 10

**Comment by OmarSultan85** [comment]
Were you able to fix this? I'm using Windows 11 and facing the same problem

**Comment by ccrowner** [comment]
No. I saw that other python packages encounter same problem on Windows.
The developers have fixed it in some of the cases. Hopefully this will be
the case for Marvin so
Windows users won't need to implement workarounds or simply not use the
package.

**Comment by andrew-healey** [comment]
I hackily fixed this on Windows 11 by modifying `marvin/utilities/async_utils.py`. New contents of the file:
```py
async def run_async(func, *args, **kwargs):
	return func(*args,**kwargs)


async def run_async_process(func, *args, **kwargs):
    # in test mode, don't spawn processes
    # if marvin.settings.test_mode:
    return await run_async(func, *args, **kwargs)
```

Then I got a `sqlalchemy` error: `sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) unable to open database file`
I fixed this by replacing my import
```py
from marvin import ai_fn
```
with the following:
```py
import marvin
from marvin import ai_fn
marvin.settings.database_connection_url = "sqlite+aiosqlite:///marvinTest.sqlite"
```

Then it worked for me.

**Comment by jlowin** [comment]
Thanks for raising this, everyone - sounds like a cross-platform issue we need to investigate a little further. Appreciate the help debugging!

**Comment by hellomusa** [comment]
Windows 10 fix that works for me: 
- In `utlities/async_utils.py`, change `concurrent.futures.ProcessPoolExecutor(mp_context=mp.get_context("fork"))` to use `fork` instead of `spawn` 
- In `config.py` (Settings), remove extra '/' in `database_connection_url`, i.e. `"sqlite+aiosqlite:///$MARVIN_HOME/marvin.sqlite"` instead of `"sqlite+aiosqlite:////$MARVIN_HOME/marvin.sqlite"`

**Comment by yichenchong** [comment]
> Windows 10 fix that works for me:
> 
> * In `utilities/async_utils.py`, change `concurrent.futures.ProcessPoolExecutor(mp_context=mp.get_context("fork"))` to use `spawn` instead of `fork`
> * In `config.py` (Settings), remove extra '/' in `database_connection_url`, i.e. `"sqlite+aiosqlite:///$MARVIN_HOME/marvin.sqlite"` instead of `"sqlite+aiosqlite:////$MARVIN_HOME/marvin.sqlite"`

Yep, can confirm this works for me as well, and was just about to post about this.

**Comment by ccrowner** [comment]
It works!

**Comment by jlowin** [comment]
I believe this was fixed in #172 - please reopen if not!
